include/seqan3/range/view/translation.hpp

```diff
@@ -0,0 +1,350 @@
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \ingroup view
+ * \author Sara Hetzel <sara.hetzel AT fu-berlin.de>
+ * \brief Provides seqan3::view::translate, seqan3::view::translate_single and seqan3::view::translate_frames.
+ */
+
+#pragma once
+
+#include <vector>
+#include <stdexcept>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/aminoacid/aa27.hpp>
+#include <seqan3/alphabet/translation/translation.hpp>
+#include <seqan3/alphabet/translation/translation_details.hpp>
+#include <seqan3/range/concept.hpp>
+#include <seqan3/range/view/complement.hpp>
+
+#include <range/v3/view/chunk.hpp>
+#include <range/v3/view/concat.hpp>
+#include <range/v3/view/drop.hpp>
+#include <range/v3/view/reverse.hpp>
+#include <range/v3/view/single.hpp>
+#include <range/v3/view/take_while.hpp>
+#include <range/v3/view/transform.hpp>
+
+namespace seqan3::detail
+{
+//!\privatesection
+/*!\brief Lambda for translation of one nucleotide triplet into a single aminoacid.
+ * \details Used by view::translate_single.
+ */
+auto translate = [] (auto triplet)
+{
+    return translate_triplet(triplet[0], triplet[1], triplet[2]);
+};
+
+struct translate_single_fn
+{
+    /*!\brief Single frame translation from nucleotide to aminoacid.
+     * \tparam irng_t The type of the range being processed.
+     * \param irange The range being processed.
+     */
+    template <typename irng_t>
+        requires input_range_concept<irng_t>/* && nucleotide_concept<ranges::value_type_t<std::decay_t<irng_t>>>*/
+    auto operator()(irng_t && irange) const
+    {
+        auto single_frame = irange |ranges::view::chunk(3)
+                                   | ranges::view::take_while([] (auto chnk) { return ranges::size(chnk) == 3; })
+                                   | ranges::view::transform(translate);
+        return single_frame;
+    }
+
+    /*!\brief Single frame translation from nucleotide to aminoacid.
+     * \tparam irng_t The type of the range being processed.
+     * \param irange The range being processed.
+     */
+    template <typename irng_t>
+        requires input_range_concept<irng_t>
+                 && sized_range_concept<irng_t>
+//                  && nucleotide_concept<ranges::value_type_t<std::decay_t<irng_t>>>
+    auto operator()(irng_t && irange) const
+    {
+        auto single_frame = irange | ranges::view::take(ranges::size(irange) / 3 * 3)
+                                   | ranges::view::chunk(3)
+                                   | ranges::view::transform(translate);
+        return single_frame;
+    }
```

Comment:
you can use 
```cpp

if constexpr (sized_range_concept<irng_t>)
{
    return irange | ranges::view::take_exactly(ranges::size(irange) / 3 * 3)
                 | ranges::view::chunk(3)
                 | ranges::view::transform(translate);
} else
{
    return irange  | ranges::view::chunk(3)
                 | ranges::view::take_while([] (auto chnk) { return ranges::size(chnk) == 3; })
                 | ranges::view::transform(translate);
}
```
This works in contrast to other case, because if you use `if constexpr` the "decision" happens at compile time and thereby the return type is unambiguous. The other case was a run-time decision which would have resulted in an ambiguous return type,